File: csharp/CreoleCentricAPI.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Threading;
using System.Threading.Tasks;

namespace CreoleCentric.Api
{
    /// <summary>
    /// CreoleCentric TTS API Client
    ///
    /// A .NET client for the CreoleCentric Text-to-Speech API.
    /// Requires .NET 5.0+ or .NET Core 3.1+
    ///
    /// Example usage:
    /// <code>
    /// var client = new CreoleCentricAPI("cc_your_api_key_here");
    /// var health = await client.CheckHealthAsync();
    /// Console.WriteLine($"API Version: {health.Version}");
    /// </code>
    /// </summary>
    public class CreoleCentricAPI : IDisposable
    {
        private readonly HttpClient _httpClient;
        private readonly string _apiKey;
        private readonly string _baseUrl;
        private readonly JsonSerializerOptions _jsonOptions;
        private bool _disposed;

        // Response models
        public class HealthCheckResponse
        {
            [JsonPropertyName("status")]
            public string Status { get; set; }

            [JsonPropertyName("service")]
            public string Service { get; set; }

            [JsonPropertyName("database")]
            public string Database { get; set; }

            [JsonPropertyName("version")]
            public string Version { get; set; }
        }

        public class CreditBalanceResponse
        {
            [JsonPropertyName("total_credits")]
            public double TotalCredits { get; set; }

            [JsonPropertyName("used_credits")]
            public double UsedCredits { get; set; }

            [JsonPropertyName("available_credits")]
            public double AvailableCredits { get; set; }

            [JsonPropertyName("currency")]
            public string Currency { get; set; }
        }

        public class Voice
        {
            [JsonPropertyName("voice_id")]
            public string VoiceId { get; set; }

            [JsonPropertyName("name")]
            public string Name { get; set; }

            [JsonPropertyName("region")]
            public string Region { get; set; }

            [JsonPropertyName("gender")]
            public string Gender { get; set; }

            [JsonPropertyName("description")]
            public string Description { get; set; }

            [JsonPropertyName("preview_url")]
            public string PreviewUrl { get; set; }

            [JsonPropertyName("is_active")]
            public bool IsActive { get; set; }
        }

        public class VoicesResponse
        {
            [JsonPropertyName("success")]
            public bool Success { get; set; }

            [JsonPropertyName("voices")]
            public List<Voice> Voices { get; set; }

            [JsonPropertyName("count")]
            public int Count { get; set; }

            [JsonPropertyName("source")]
            public string Source { get; set; }
        }

        public class Model
        {
            [JsonPropertyName("id")]
            public string Id { get; set; }

            [JsonPropertyName("name")]
            public string Name { get; set; }

            [JsonPropertyName("description")]
            public string Description { get; set; }

            [JsonPropertyName("version")]
            public string Version { get; set; }

            [JsonPropertyName("is_active")]
            public bool IsActive { get; set; }

            [JsonPropertyName("supports_streaming")]
            public bool SupportsStreaming { get; set; }
        }

        public class ModelsResponse
        {
            [JsonPropertyName("success")]
            public bool Success { get; set; }

            [JsonPropertyName("models")]
            public List<Model> Models { get; set; }

            [JsonPropertyName("count")]
            public int Count { get; set; }
        }

        public class TTSJob
        {
            [JsonPropertyName("id")]
            public string Id { get; set; }

            [JsonPropertyName("status")]
            public string Status { get; set; }

            [JsonPropertyName("text_input")]
            public string TextInput { get; set; }

            [JsonPropertyName("voice_id")]
            public string VoiceId { get; set; }

            [JsonPropertyName("model_id")]
            public string ModelId { get; set; }

            [JsonPropertyName("created_at")]
            public string CreatedAt { get; set; }
        }

        public class JobStatusResponse
        {
            [JsonPropertyName("id")]
            public string Id { get; set; }

            [JsonPropertyName("status")]
            public string Status { get; set; }

            [JsonPropertyName("audio_url")]
            public string AudioUrl { get; set; }

            [JsonPropertyName("created_at")]
            public string CreatedAt { get; set; }

            [JsonPropertyName("updated_at")]
            public string UpdatedAt { get; set; }

            [JsonPropertyName("error")]
            public string Error { get; set; }
        }

        public class RecentJobsResponse
        {
            [JsonPropertyName("success")]
            public bool Success { get; set; }

            [JsonPropertyName("jobs")]
            public List<JobStatusResponse> Jobs { get; set; }

            [JsonPropertyName("count")]
            public int Count { get; set; }
        }

        /// <summary>
        /// Initialize the CreoleCentric API client
        /// </summary>
        /// <param name="apiKey">Your CreoleCentric API key (starts with 'cc_')</param>
        public CreoleCentricAPI(string apiKey)
            : this(apiKey, "https://creolecentric.com/api/v1")
        {
        }

        /// <summary>
        /// Initialize the CreoleCentric API client with custom base URL
        /// </summary>
        /// <param name="apiKey">Your CreoleCentric API key</param>
        /// <param name="baseUrl">Custom base URL for the API</param>
        public CreoleCentricAPI(string apiKey, string baseUrl)
        {
            _apiKey = apiKey ?? throw new ArgumentNullException(nameof(apiKey));
            _baseUrl = baseUrl ?? throw new ArgumentNullException(nameof(baseUrl));

            _httpClient = new HttpClient
            {
                BaseAddress = new Uri(_baseUrl),
                Timeout = TimeSpan.FromSeconds(30)
            };

            _httpClient.DefaultRequestHeaders.Authorization =
                new AuthenticationHeaderValue("Bearer", _apiKey);
            _httpClient.DefaultRequestHeaders.Accept.Add(
                new MediaTypeWithQualityHeaderValue("application/json"));

            _jsonOptions = new JsonSerializerOptions
            {
                PropertyNameCaseInsensitive = true,
                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
            };
        }

        /// <summary>
        /// Make an HTTP request to the API
        /// </summary>
        private async Task<T> MakeRequestAsync<T>(
            HttpMethod method,
            string endpoint,
            object body = null,
            CancellationToken cancellationToken = default)
        {
            var request = new HttpRequestMessage(method, endpoint);

            if (body != null && method == HttpMethod.Post)
            {
                var json = JsonSerializer.Serialize(body, _jsonOptions);
                request.Content = new StringContent(json, Encoding.UTF8, "application/json");
            }

            var response = await _httpClient.SendAsync(request, cancellationToken);

            if (!response.IsSuccessStatusCode)
            {
                var errorContent = await response.Content.ReadAsStringAsync();
                throw new HttpRequestException(
                    $"HTTP {(int)response.StatusCode}: {errorContent}");
            }

            var responseContent = await response.Content.ReadAsStringAsync();
            return JsonSerializer.Deserialize<T>(responseContent, _jsonOptions);
        }

        /// <summary>
        /// Check API health and get version information
        /// </summary>
        public Task<HealthCheckResponse> CheckHealthAsync(
            CancellationToken cancellationToken = default)
        {
            return MakeRequestAsync<HealthCheckResponse>(
                HttpMethod.Get, "/health/", null, cancellationToken);
        }

        /// <summary>
        /// Get current credit balance
        /// </summary>
        public Task<CreditBalanceResponse> GetCreditBalanceAsync(
            CancellationToken cancellationToken = default)
        {
            return MakeRequestAsync<CreditBalanceResponse>(
                HttpMethod.Get, "/credits/balance/", null, cancellationToken);
        }

        /// <summary>
        /// Get list of available voices
        /// </summary>
        public Task<VoicesResponse> GetVoicesAsync(
            CancellationToken cancellationToken = default)
        {
            return MakeRequestAsync<VoicesResponse>(
                HttpMethod.Get, "/tts/voices/", null, cancellationToken);
        }

        /// <summary>
        /// Get list of available models
        /// </summary>
        public Task<ModelsResponse> GetModelsAsync(
            CancellationToken cancellationToken = default)
        {
            return MakeRequestAsync<ModelsResponse>(
                HttpMethod.Get, "/tts/models/", null, cancellationToken);
        }

        /// <summary>
        /// Create a new TTS job
        /// </summary>
        /// <param name="text">The text to convert to speech</param>
        /// <param name="voiceId">The ID of the voice to use</param>
        /// <param name="modelId">The ID of the model to use</param>
        /// <param name="speed">Speech speed (0.5 to 2.0)</param>
        /// <param name="stability">Voice stability (0.0 to 1.0)</param>
        /// <param name="similarityBoost">Voice similarity boost (0.0 to 1.0)</param>
        /// <param name="cancellationToken">Cancellation token</param>
        public Task<TTSJob> CreateTTSJobAsync(
            string text,
            string voiceId,
            string modelId,
            double speed = 1.0,
            double stability = 0.5,
            double similarityBoost = 0.75,
            CancellationToken cancellationToken = default)
        {
            var payload = new
            {
                text,
                voice_id = voiceId,
                model_id = modelId,
                speed,
                stability,
                similarity_boost = similarityBoost
            };

            return MakeRequestAsync<TTSJob>(
                HttpMethod.Post, "/tts/submit/", payload, cancellationToken);
        }

        /// <summary>
        /// Get the status of a TTS job
        /// </summary>
        /// <param name="jobId">The job ID</param>
        /// <param name="cancellationToken">Cancellation token</param>
        public Task<JobStatusResponse> GetJobStatusAsync(
            string jobId,
            CancellationToken cancellationToken = default)
        {
            return MakeRequestAsync<JobStatusResponse>(
                HttpMethod.Get, $"/tts/job/{jobId}/", null, cancellationToken);
        }

        /// <summary>
        /// Wait for a job to complete
        /// </summary>
        /// <param name="jobId">The job ID</param>
        /// <param name="timeout">Maximum time to wait in seconds</param>
        /// <param name="pollInterval">How often to check status in seconds</param>
        /// <param name="cancellationToken">Cancellation token</param>
        public async Task<JobStatusResponse> WaitForJobAsync(
            string jobId,
            int timeout = 300,
            int pollInterval = 2,
            CancellationToken cancellationToken = default)
        {
            var startTime = DateTime.UtcNow;
            var timeoutSpan = TimeSpan.FromSeconds(timeout);
            var pollIntervalSpan = TimeSpan.FromSeconds(pollInterval);

            while (DateTime.UtcNow - startTime < timeoutSpan)
            {
                var status = await GetJobStatusAsync(jobId, cancellationToken);

                if (status.Status == "completed" ||
                    status.Status == "delivered" ||
                    status.Status == "failed" ||
                    status.Status == "cancelled")
                {
                    return status;
                }

                Console.WriteLine($"Job {jobId} status: {status.Status}");
                await Task.Delay(pollIntervalSpan, cancellationToken);
            }

            throw new TimeoutException(
                $"Job {jobId} did not complete within {timeout} seconds");
        }

        /// <summary>
        /// Get recent TTS jobs
        /// </summary>
        /// <param name="limit">Maximum number of jobs to return</param>
        /// <param name="cancellationToken">Cancellation token</param>
        public Task<RecentJobsResponse> GetRecentJobsAsync(
            int limit = 10,
            CancellationToken cancellationToken = default)
        {
            return MakeRequestAsync<RecentJobsResponse>(
                HttpMethod.Get, $"/tts/jobs/?limit={limit}", null, cancellationToken);
        }

        /// <summary>
        /// Dispose of the HTTP client
        /// </summary>
        public void Dispose()
        {
            Dispose(true);
            GC.SuppressFinalize(this);
        }

        protected virtual void Dispose(bool disposing)
        {
            if (!_disposed)
            {
                if (disposing)
                {
                    _httpClient?.Dispose();
                }
                _disposed = true;
            }
        }

        /// <summary>
        /// Example usage demonstrating all features
        /// </summary>
        public static async Task Main(string[] args)
        {
            var apiKey = Environment.GetEnvironmentVariable("CREOLECENTRIC_API_KEY");
            if (string.IsNullOrEmpty(apiKey))
            {
                Console.WriteLine("Error: CREOLECENTRIC_API_KEY environment variable not set");
                Environment.Exit(1);
            }

            using var client = new CreoleCentricAPI(apiKey);

            try
            {
                // 1. Health Check
                Console.WriteLine("=== Health Check ===");
                var health = await client.CheckHealthAsync();
                Console.WriteLine($"Status: {health.Status}");
                Console.WriteLine($"Service: {health.Service}");
                Console.WriteLine($"Version: {health.Version}");
                Console.WriteLine();

                // 2. Credit Balance
                Console.WriteLine("=== Credit Balance ===");
                var balance = await client.GetCreditBalanceAsync();
                Console.WriteLine($"Total Credits: {balance.TotalCredits}");
                Console.WriteLine($"Used Credits: {balance.UsedCredits}");
                Console.WriteLine($"Available Credits: {balance.AvailableCredits}");
                Console.WriteLine();

                // 3. List Voices
                Console.WriteLine("=== Available Voices ===");
                var voicesResponse = await client.GetVoicesAsync();
                var voices = voicesResponse.Voices;
                Console.WriteLine($"Found {voicesResponse.Count} voices (source: {voicesResponse.Source}):");
                for (int i = 0; i < Math.Min(5, voices.Count); i++)
                {
                    var voice = voices[i];
                    Console.WriteLine($"  - {voice.Name} ({voice.VoiceId})");
                    Console.WriteLine($"    Region: {voice.Region}, Gender: {voice.Gender}");
                }
                Console.WriteLine();

                // 4. List Models
                Console.WriteLine("=== Available Models ===");
                var modelsResponse = await client.GetModelsAsync();
                var models = modelsResponse.Models;
                Console.WriteLine($"Found {modelsResponse.Count} models:");
                foreach (var model in models)
                {
                    Console.WriteLine($"  - {model.Name} ({model.Id})");
                    Console.WriteLine($"    {model.Description}");
                }
                Console.WriteLine();

                // 5. Create TTS Job
                var voiceId = "i4mRPwKM2yHwXhbmkN514";  // Xavier Bruneau
                var modelId = "ccl_ht_v100";

                if (voices.Count > 0 &&
                    voices[0].VoiceId != "voice_1" &&
                    voices[0].VoiceId != "voice_2")
                {
                    voiceId = voices[0].VoiceId;
                }
                if (models.Count > 0)
                {
                    modelId = models[0].Id;
                }

                Console.WriteLine("=== Creating TTS Job ===");
                var text = "Bonjou! Sa se yon egzanp nan itilizasyon API CreoleCentric pou konvèti tèks an Kreyòl Ayisyen an lapawòl.";
                Console.WriteLine($"Text: {text}");
                Console.WriteLine($"Voice ID: {voiceId}");
                Console.WriteLine($"Model ID: {modelId}");

                var job = await client.CreateTTSJobAsync(text, voiceId, modelId);
                Console.WriteLine($"Job created: {job.Id}");
                Console.WriteLine($"Status: {job.Status}");
                Console.WriteLine();

                // 6. Wait for Completion
                Console.WriteLine("=== Waiting for Job Completion ===");
                var finalStatus = await client.WaitForJobAsync(job.Id, 60, 2);
                Console.WriteLine($"Job {finalStatus.Id} {finalStatus.Status}");

                if (finalStatus.Status == "completed" || finalStatus.Status == "delivered")
                {
                    Console.WriteLine($"Audio URL: {finalStatus.AudioUrl}");
                }
                else if (finalStatus.Status == "failed")
                {
                    Console.WriteLine($"Error: {finalStatus.Error}");
                }
                Console.WriteLine();

                // 7. Recent Jobs
                Console.WriteLine("=== Recent Jobs ===");
                var recentJobs = await client.GetRecentJobsAsync(5);
                Console.WriteLine($"Found {recentJobs.Count} recent jobs:");
                foreach (var recentJob in recentJobs.Jobs)
                {
                    Console.WriteLine($"  - Job {recentJob.Id}: {recentJob.Status}");
                    if (!string.IsNullOrEmpty(recentJob.AudioUrl))
                    {
                        Console.WriteLine($"    Audio: {recentJob.AudioUrl}");
                    }
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
                Environment.Exit(1);
            }
        }
    }
}

File: swift/CreoleCentricAPI.swift
import Foundation

// MARK: - Type Definitions

struct HealthResponse: Codable {
    let status: String
    let service: String
    let database: String
    let version: String
}

struct CreditBalance: Codable {
    let totalCredits: Int
    let subscriptionCredits: Int
    let purchasedCredits: Int
    let subscriptionPlan: String?

    enum CodingKeys: String, CodingKey {
        case totalCredits = "total_credits"
        case subscriptionCredits = "subscription_credits"
        case purchasedCredits = "purchased_credits"
        case subscriptionPlan = "subscription_plan"
    }
}

struct Voice: Codable {
    let id: String
    let voiceId: String
    let name: String
    let description: String
    let gender: String
    let region: String
    let regionCode: String?
    let accent: String?
    let style: String?
    let qualityLevel: String?
    let ageGroup: String?
    let languageCode: String
    let isActive: Bool
    let isDefault: Bool
    let isFeatured: Bool
    let isPremium: Bool
    let previewUrl: String?
    let previewText: String?
    let tags: [String]?
    let defaultStability: Double?
    let defaultSimilarityBoost: Double?

    enum CodingKeys: String, CodingKey {
        case id, name, description, gender, region, accent, style, tags
        case voiceId = "voice_id"
        case regionCode = "region_code"
        case qualityLevel = "quality_level"
        case ageGroup = "age_group"
        case languageCode = "language_code"
        case isActive = "is_active"
        case isDefault = "is_default"
        case isFeatured = "is_featured"
        case isPremium = "is_premium"
        case previewUrl = "preview_url"
        case previewText = "preview_text"
        case defaultStability = "default_stability"
        case defaultSimilarityBoost = "default_similarity_boost"
    }
}

struct VoicesResponse: Codable {
    let success: Bool
    let voices: [Voice]
    let count: Int
    let source: String
}

struct Model: Codable {
    let id: String
    let name: String
    let displayName: String?
    let description: String
    let version: String?
    let qualityScore: Double?
    let isDefault: Bool?
    let isPremium: Bool?
    let supportedLanguages: [String]?

    enum CodingKeys: String, CodingKey {
        case id, name, description, version
        case displayName = "display_name"
        case qualityScore = "quality_score"
        case isDefault = "is_default"
        case isPremium = "is_premium"
        case supportedLanguages = "supported_languages"
    }
}

struct ModelsResponse: Codable {
    let success: Bool
    let models: [Model]
    let count: Int
}

struct TTSJob: Codable {
    let id: String
    let text: String
    let voiceId: String
    let voiceName: String?
    let modelId: String
    let status: String
    let createdAt: String
    let startedAt: String?
    let completedAt: String?
    let audioFileUrl: String?
    let durationSeconds: Double?
    let durationFormatted: String?
    let creditsUsed: Int
    let charactersProcessed: Int?
    let errorMessage: String?

    enum CodingKeys: String, CodingKey {
        case id, text, status
        case voiceId = "voice_id"
        case voiceName = "voice_name"
        case modelId = "model_id"
        case createdAt = "created_at"
        case startedAt = "started_at"
        case completedAt = "completed_at"
        case audioFileUrl = "audio_file_url"
        case durationSeconds = "duration_seconds"
        case durationFormatted = "duration_formatted"
        case creditsUsed = "credits_used"
        case charactersProcessed = "characters_processed"
        case errorMessage = "error_message"
    }
}

struct JobStatusResponse: Codable {
    let status: String
    let audioUrl: String?
    let durationSeconds: Double?
    let errorMessage: String?

    enum CodingKeys: String, CodingKey {
        case status
        case audioUrl = "audio_url"
        case durationSeconds = "duration_seconds"
        case errorMessage = "error_message"
    }
}

struct JobListResponse: Codable {
    let results: [TTSJob]
    let count: Int
    let next: String?
    let previous: String?
}

struct CreateJobRequest: Codable {
    let text: String
    let voiceId: String
    let modelId: String
    let webhookUrl: String?
    let speed: Double?
    let stability: Double?
    let similarityBoost: Double?
    let style: Double?
    let useSpeakerBoost: Bool?

    enum CodingKeys: String, CodingKey {
        case text
        case voiceId = "voice_id"
        case modelId = "model_id"
        case webhookUrl = "webhook_url"
        case speed, stability, style
        case similarityBoost = "similarity_boost"
        case useSpeakerBoost = "use_speaker_boost"
    }
}

// MARK: - API Client

enum CreoleCentricError: Error {
    case invalidURL
    case invalidResponse
    case httpError(statusCode: Int, message: String)
    case decodingError(Error)
    case networkError(Error)
    case timeout

    var localizedDescription: String {
        switch self {
        case .invalidURL:
            return "Invalid URL"
        case .invalidResponse:
            return "Invalid response from server"
        case .httpError(let statusCode, let message):
            return "HTTP \(statusCode): \(message)"
        case .decodingError(let error):
            return "Failed to decode response: \(error.localizedDescription)"
        case .networkError(let error):
            return "Network error: \(error.localizedDescription)"
        case .timeout:
            return "Request timed out"
        }
    }
}

class CreoleCentricAPI {
    private let apiKey: String
    private let baseURL: String
    private let session: URLSession

    init(apiKey: String, baseURL: String = "https://creolecentric.com/api/v1") {
        self.apiKey = apiKey
        self.baseURL = baseURL.trimmingCharacters(in: CharacterSet(charactersIn: "/"))

        let config = URLSessionConfiguration.default
        config.timeoutIntervalForRequest = 30
        config.timeoutIntervalForResource = 60
        self.session = URLSession(configuration: config)
    }

    // MARK: - Private Methods

    private func makeRequest<T: Decodable>(
        method: String,
        endpoint: String,
        body: Encodable? = nil
    ) async throws -> T {
        guard let url = URL(string: baseURL + endpoint) else {
            throw CreoleCentricError.invalidURL
        }

        var request = URLRequest(url: url)
        request.httpMethod = method
        request.setValue("ApiKey \(apiKey)", forHTTPHeaderField: "Authorization")
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")

        if let body = body {
            let encoder = JSONEncoder()
            encoder.keyEncodingStrategy = .convertToSnakeCase
            request.httpBody = try encoder.encode(body)
        }

        let (data, response) = try await session.data(for: request)

        guard let httpResponse = response as? HTTPURLResponse else {
            throw CreoleCentricError.invalidResponse
        }

        guard (200...299).contains(httpResponse.statusCode) else {
            let message = String(data: data, encoding: .utf8) ?? "Unknown error"
            throw CreoleCentricError.httpError(statusCode: httpResponse.statusCode, message: message)
        }

        let decoder = JSONDecoder()
        decoder.keyDecodingStrategy = .convertFromSnakeCase

        do {
            return try decoder.decode(T.self, from: data)
        } catch {
            throw CreoleCentricError.decodingError(error)
        }
    }

    // MARK: - Health Check

    func checkHealth() async throws -> HealthResponse {
        return try await makeRequest(method: "GET", endpoint: "/health/")
    }

    // MARK: - User & Credits

    func getCreditBalance() async throws -> CreditBalance {
        return try await makeRequest(method: "GET", endpoint: "/credits/balance/")
    }

    // MARK: - Voices & Models

    func getVoices() async throws -> VoicesResponse {
        return try await makeRequest(method: "GET", endpoint: "/tts/voices/")
    }

    func getModels() async throws -> ModelsResponse {
        return try await makeRequest(method: "GET", endpoint: "/tts/models/")
    }

    // MARK: - TTS Jobs

    func createTTSJob(
        text: String,
        voiceId: String,
        modelId: String,
        speed: Double? = nil,
        stability: Double? = nil,
        similarityBoost: Double? = nil
    ) async throws -> TTSJob {
        let request = CreateJobRequest(
            text: text,
            voiceId: voiceId,
            modelId: modelId,
            webhookUrl: nil,
            speed: speed,
            stability: stability,
            similarityBoost: similarityBoost,
            style: nil,
            useSpeakerBoost: nil
        )

        return try await makeRequest(method: "POST", endpoint: "/tts/jobs/", body: request)
    }

    func getJobStatus(jobId: String) async throws -> JobStatusResponse {
        return try await makeRequest(method: "GET", endpoint: "/tts/jobs/\(jobId)/status/")
    }

    func getJobDetails(jobId: String) async throws -> TTSJob {
        return try await makeRequest(method: "GET", endpoint: "/tts/jobs/\(jobId)/")
    }

    func listJobs(limit: Int = 10, offset: Int = 0) async throws -> JobListResponse {
        return try await makeRequest(method: "GET", endpoint: "/tts/jobs/list/?limit=\(limit)&offset=\(offset)")
    }

    func cancelJob(jobId: String) async throws {
        let _: [String: String] = try await makeRequest(method: "POST", endpoint: "/tts/jobs/\(jobId)/cancel/")
    }

    // MARK: - Express TTS

    func expressTTS(text: String, voiceId: String) async throws -> Data {
        guard let url = URL(string: baseURL + "/tts/express/") else {
            throw CreoleCentricError.invalidURL
        }

        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("ApiKey \(apiKey)", forHTTPHeaderField: "Authorization")
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")

        let body = ["text": text, "voice_id": voiceId]
        request.httpBody = try JSONSerialization.data(withJSONObject: body)

        let (data, response) = try await session.data(for: request)

        guard let httpResponse = response as? HTTPURLResponse,
              (200...299).contains(httpResponse.statusCode) else {
            throw CreoleCentricError.invalidResponse
        }

        return data
    }
}

// MARK: - Example Usage (Command Line)

@main
struct CreoleCentricExample {
    static func main() async {
        guard let apiKey = ProcessInfo.processInfo.environment["CREOLECENTRIC_API_KEY"] else {
            print("Error: CREOLECENTRIC_API_KEY environment variable not set")
            print("Please set it with your API key: export CREOLECENTRIC_API_KEY='cc_your_key_here'")
            return
        }

        let client = CreoleCentricAPI(apiKey: apiKey)

        do {
            // 1. Check API health
            print(String(repeating: "=", count: 50))
            print("1. Checking API Health")
            print(String(repeating: "=", count: 50))
            let health = try await client.checkHealth()
            print("API Status: \(health.status)")
            print("Version: \(health.version)\n")

            // 2. Get credit balance
            print(String(repeating: "=", count: 50))
            print("2. Credit Balance")
            print(String(repeating: "=", count: 50))
            let balance = try await client.getCreditBalance()
            print("Total Credits: \(balance.totalCredits)")
            print("Subscription Credits: \(balance.subscriptionCredits)")
            print("Purchased Credits: \(balance.purchasedCredits)\n")

            // 3. Get available voices
            print(String(repeating: "=", count: 50))
            print("3. Available Voices")
            print(String(repeating: "=", count: 50))
            let voicesResponse = try await client.getVoices()
            let voices = voicesResponse.voices
            print("Found \(voicesResponse.count) voices (source: \(voicesResponse.source)):")
            for voice in voices.prefix(5) {
                print("  - \(voice.name) (ID: \(voice.voiceId))")
                print("    Region: \(voice.region), Gender: \(voice.gender)")
            }
            print()

            // 4. Get available models
            print(String(repeating: "=", count: 50))
            print("4. Available Models")
            print(String(repeating: "=", count: 50))
            let modelsResponse = try await client.getModels()
            let models = modelsResponse.models
            print("Found \(modelsResponse.count) models:")
            for model in models {
                let displayName = model.displayName ?? model.name
                print("  - \(displayName) (ID: \(model.id))")
                print("    Description: \(model.description)")
            }
            print()

            // 5. Create a TTS job
            print(String(repeating: "=", count: 50))
            print("5. Creating TTS Job")
            print(String(repeating: "=", count: 50))

            let text = "Bonjou! Mwen se yon egzanp API pou CreoleCentric. Mwen ka pale Kreyòl ayisyen."
            print("Text: \(text)")

            var voiceId = "i4mRPwKM2yHwXhbmkN514"  // Xavier Bruneau
            var modelId = "ccl_ht_v100"

            // Use first voice if not a placeholder
            if let firstVoice = voices.first,
               !["voice_1", "voice_2"].contains(firstVoice.voiceId) {
                voiceId = firstVoice.voiceId
            }
            if let firstModel = models.first {
                modelId = firstModel.id
            }

            let job = try await client.createTTSJob(
                text: text,
                voiceId: voiceId,
                modelId: modelId
            )

            print("Job created successfully!")
            print("Job ID: \(job.id)")
            print("Status: \(job.status)")
            print("Credits used: \(job.creditsUsed)\n")
            print("📢 To receive webhook notifications, pass webhookUrl parameter to createTTSJob:")
            print("   Events: tts_queued → tts_started → tts_synthesized → tts_uploaded → tts_delivered")
            print("   See examples/webhook_server.swift for webhook handling example\n")

            // 6. List recent jobs
            print(String(repeating: "=", count: 50))
            print("6. Recent Jobs")
            print(String(repeating: "=", count: 50))

            let jobList = try await client.listJobs(limit: 5)
            print("Recent \(jobList.results.count) jobs:")

            for job in jobList.results {
                let jobIdDisplay = String(job.id.prefix(8)) + "..."
                print("  - Job \(jobIdDisplay)")
                print("    Created: \(job.createdAt)")
                print("    Status: \(job.status)")
                if !job.text.isEmpty {
                    let textPreview = String(job.text.prefix(50))
                    print("    Text: \(textPreview)\(job.text.count > 50 ? "..." : "")")
                }
            }

        } catch {
            print("Error: \(error.localizedDescription)")
        }
    }
}
